src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryPath.java

```diff
@@ -88,9 +99,14 @@ private void validateName(String property, String name) {
       throw new IllegalArgumentException(
         String.format("%s is invalid, it should contain at most %d characters.", property, NAME_MAX_LENGTH));
     }
-    if (!name.matches(VALID_NAME_REGEX)) {
+    if (property.equals("Dataset") && !VALID_DATASET_NAME_REGEX.matcher(name).matches()) {
```

Comment:
Where the `"Dataset"` comes from? Is it some hardcoded default? Same for the `"Table"` below

Reply:
Yeah, it's hardcoded and used whenever this `validateName` function is called.
For example, in [DBPath:79](https://github.com/cdapio/hydrator-plugins/blob/2ae29b018990e419f726adcf8df50a75759be3f5/hydrator-common/src/main/java/io/cdap/plugin/common/db/DBPath.java#L79) and [DBSpecificPath:90](https://github.com/data-integrations/database-plugins/blob/ee8eda9ee2631073eecea72899bc95d72f0b8637/database-commons/src/main/java/io/cdap/plugin/db/connector/DBSpecificPath.java#L90)